AdyenCard/Utilities/CardType/CardBrandProvider.swift

```diff
@@ -103,7 +108,7 @@ internal final class CardBrandProvider: AnyCardBrandProvider {
             return success(binLookupService)
         }
         
-        let localApiClient = self.apiClient ?? APIClient(environment: self.environment)
+        let localApiClient = self.apiClient ?? APIClient(apiContext: apiContext)
```

Comment:
In this class we only use `apiContext: apiContext` to create `APIClient` if it was not provided on `init` - maybe just change init ?

```swift
internal init(cardPublicKeyProvider: AnyCardPublicKeyProvider,
              apiClient: APIClientProtocol,
              fallbackCardTypeProvider: AnyCardBrandProvider = FallbackCardBrandProvider()) { ... }
```

Reply:
I like this idea. But we need both `apiClient` and `apiContext` for tests, so they both need to be passed into the init. But we can make `apiClient` non-optional and construct it in `init`.